player/cPlayer.h

```diff
@@ -51,7 +53,7 @@ struct s_PlaceResult {
 class cPlayer : public cScenarioObserver {
 
 public:
-    static const int MAX_STRUCTURE_BMPS = MAX_STRUCTURETYPES*2;
+    static constexpr int MAX_STRUCTURE_BMPS = MAX_STRUCTURETYPES*2;
```

Comment:
what is the difference between `const` and `constexpr` ? A quick google tells me it has to do with when this is initialized (compile time vs runtime). Thats it? Is it a best practice to use this instead of `const` ?

Reply:
constexpr Allows variables and functions to be evaluated at compile-time. In this case, the `MAX_STRUCTURETYPES*2` is calculated by the compiler and not by the program when it's initialized. Doesn't make that much difference in this case, but it is indeed a best practice. Because we can now use this variable in other constexpr functions/expressions and they will then also be evaluated at compile-time.

Reply:
I also think the optimizer can throw the variable out if it's not used at runtime anymore. So there would be no memory reserved for this integer. Not 100% sure, though. (As one can never be with the optimizer. ;) )